verticapy/learn/vmodel.py

```diff
@@ -2502,15 +2516,30 @@ def fit(
                 input_relation = vdf_from_relation(input_relation)
             input_relation.astype(new_types)
         does_model_exist(name=self.name, raise_error=True)
-        if isinstance(input_relation, vDataFrame):
-            self.input_relation = input_relation.__genSQL__()
+        id_column, id_column_name = "", gen_tmp_name(name="id_column")
+        if self.type in (
+            "RandomForestClassifier",
+            "RandomForestRegressor",
+            "XGBoostClassifier",
+            "XGBoostRegressor",
+        ) and isinstance(verticapy.options["random_state"], int):
+            id_column = ", ROW_NUMBER() OVER (ORDER BY {0}) AS {1}".format(
+                ", ".join(X), id_column_name
+            )
+        tmp_view = False
+        if isinstance(input_relation, vDataFrame) or (id_column):
+            tmp_view = True
+            if isinstance(input_relation, vDataFrame):
+                self.input_relation = input_relation.__genSQL__()
+            else:
+                self.input_relation = input_relation
             relation = gen_tmp_name(schema=schema_relation(self.name)[0], name="view")
             drop_if_exists(relation, method="view")
             executeSQL(
-                "CREATE VIEW {} AS SELECT * FROM {}".format(
-                    relation, input_relation.__genSQL__()
+                "CREATE VIEW {0} AS SELECT *{1} FROM {2}".format(
+                    relation, id_column, self.input_relation
                 ),
-                print_time_sql=False,
+                title="Creating a temporary view to fit the model.",
```

Comment:
IIUC, you create an id_column on the fly in a temporary view to train the model. Please note that the training will be fixed on this particular view. Can you guarantee that the row numbers would be exactly the same next time that you create the view?

Reply:
Yes as it is ordered by all the columns in the dataset.